modules/api/src/main/java/org/apache/fluo/api/observer/MetricsReporter.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.fluo.api.observer;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Reports application metrics using Fluo metrics reporters configured by 'fluo.metrics.reporter.*'
+ * properties. Use unique names when naming metrics.
+ *
+ * @since 1.0.0
+ */
+public interface MetricsReporter {
```

Comment:
Was thinking about API and was wondering if it would be better to structure API like following.

```java
MetricsReporter reporter = ...
Counter counter = reporter.counter("foo");
counter.update(9);
```

This allows potential to keep a refernce to a metric for repeated use and avoid looking up the metric by name.